File: Sources/PhotoPullCore/TransferMode.swift
import Foundation

/// Режим переноса медиафайлов с устройства.
public enum TransferMode: String, Codable, CaseIterable, Sendable {
    /// Копировать: файлы остаются на устройстве.
    case copy
    /// Переместить: после успешной загрузки файл удаляется с устройства.
    case move

    /// Человекочитаемое название для UI.
    public var title: String {
        switch self {
        case .copy: return "Копировать"
        case .move: return "Переместить"
        }
    }

    /// Нужно ли удалять оригинал на устройстве после успешной загрузки.
    ///
    /// Транслируется в опцию `ICDeleteAfterSuccessfulDownload` при вызове
    /// `requestDownloadFile(_:options:...)`.
    public var deletesSourceAfterDownload: Bool {
        self == .move
    }
}

File: shared/src/commonMain/kotlin/com/async/shared/TransferMode.kt
package com.async.shared

/**
 * Режим переноса медиафайлов с устройства.
 */
enum class TransferMode(
    /** Стабильный строковый идентификатор для персистентности/интеропа. */
    val id: String,
    /** Человекочитаемое название для UI. */
    val title: String
) {
    /** Копировать: файлы остаются на устройстве. */
    COPY(id = "copy", title = "Копировать"),

    /** Переместить: после успешной загрузки файл удаляется с устройства. */
    MOVE(id = "move", title = "Переместить");

    /**
     * Нужно ли удалять оригинал на устройстве после успешной загрузки.
     *
     * На стороне macOS транслируется в опцию `ICDeleteAfterSuccessfulDownload`.
     */
    val deletesSourceAfterDownload: Boolean
        get() = this == MOVE

    companion object {
        /** Разбор режима по строковому [id]; по умолчанию — [COPY]. */
        fun fromId(id: String?): TransferMode =
            entries.firstOrNull { it.id == id } ?: COPY
    }
}
